crates/codegen/tests/integration_test.rs

```diff
@@ -0,0 +1,29 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use javy_codegen::{Generator, LinkingKind, Plugin, JS};
+
+#[test]
+fn test_empty() -> Result<()> {
+    // Load valid JS from file.
+    let js = JS::from_file(
+        PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("tests")
+            .join("sample-scripts")
+            .join("empty.js")
+            .as_path(),
+    )?;
+
+    // Load existing pre-initialized Javy plugin.
+    let plugin =
+        Plugin::new_from_path(PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test_plugin.wasm"))?;
+
+    // Configure code generator.
+    let mut generator = Generator::new(plugin);
+    generator.linking(LinkingKind::Static);
+
+    // Generate valid WASM module.
+    generator.generate(&js)?;
```

Comment:
I guess playing off the comment above, can we validate the returned vector to ensure it returned valid Wasm bytecode when there are no errors?

Reply:
Are you interested in verifying all the exports like in the `UninitializedPlugin` struct? (In which case we should move that over to this crate and make it public). Or is just a simple validation sufficient like below? 

E.X. Simple verification.

```rust
    // Generate valid Wasm module.
    let wasm = generator.generate(&js)?;

    // Verify generated bytes are valid Wasm.
    walrus::Module::from_buffer(wasm.as_slice())?;
```

Reply:
Just a simple verification is fine! Just a sanity check to ensure it's not returning an empty vector or some other byte buffer.